src/client/mod.rs

```diff
@@ -0,0 +1,66 @@
+use crate::{
+    pool::{
+        mempool::{KakarotPool, TransactionOrdering},
+        validate::KakarotTransactionValidatorBuilder,
+    },
+    providers::eth_provider::{
+        database::{state::EthDatabase, Database},
+        error::KakarotError,
+        provider::EthDataProvider,
+    },
+};
+use num_traits::ToPrimitive;
+use reth_chainspec::ChainSpec;
+use reth_transaction_pool::{blobstore::NoopBlobStore, EthPooledTransaction, PoolConfig};
+use starknet::{core::types::Felt, providers::Provider};
+use std::sync::Arc;
+
+/// Provides a wrapper structure around the Ethereum Provider
+/// and the Mempool.
+#[derive(Debug, Clone)]
+pub struct EthClient<SP: Provider + Send + Sync> {
+    eth_provider: EthDataProvider<SP>,
+    pool: Arc<KakarotPool<EthDataProvider<SP>>>,
+}
+
+impl<SP> EthClient<SP>
+where
+    SP: Provider + Clone + Sync + Send,
+{
+    /// Tries to start a [`EthClient`] by fetching the current chain id, initializing a [`EthDataProvider`] and
+    /// a `Pool`.
+    pub async fn try_new(sn_provider: SP, database: Database) -> eyre::Result<Self> {
```

Comment:
Instead of `sn_provider`, we could use `starknet_provider` to be consistent with the other places where we use it. What do you think?